File: backend/src/main/java/com/yourorg/appname/service/StudentService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.request.StudentRequest;
import com.yourorg.appname.dto.response.StudentResponse;
import java.util.List;

public interface StudentService {
    List<StudentResponse> getAllStudents(String query, String feeStatus, String wing);
    StudentResponse getStudentById(Long id);
    StudentResponse getStudentByRollNumber(String rollNumber);
    StudentResponse createStudent(StudentRequest request);
    StudentResponse updateStudent(Long id, StudentRequest request);
    void deleteStudent(Long id);
}

File: frontend/src/services/studentService.js
import apiClient from './apiClient';

export const studentService = {
  getAllStudents: async (params = {}) => {
    const response = await apiClient.get('/api/students', { params });
    return response.data;
  },
  getStudentById: async (id) => {
    const response = await apiClient.get(`/api/students/${id}`);
    return response.data;
  },
  createStudent: async (data) => {
    const response = await apiClient.post('/api/students', data);
    return response.data;
  },
  updateStudent: async (id, data) => {
    const response = await apiClient.put(`/api/students/${id}`, data);
    return response.data;
  },
  deleteStudent: async (id) => {
    const response = await apiClient.delete(`/api/students/${id}`);
    return response.data;
  },
};
